cli/src/invite_device.rs

```diff
@@ -0,0 +1,118 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use clap::Args;
+use std::path::PathBuf;
+use terminal_spinners::{SpinnerBuilder, DOTS};
+
+use libparsec::{
+    authenticated_cmds::latest::invite_new::{self, InviteNewRep, UserOrDevice},
+    get_default_config_dir, list_available_devices, load_device, AuthenticatedCmds,
+    BackendInvitationAddr, DeviceAccessStrategy, DeviceFileType, InvitationType, ProxyConfig,
+};
+
+use crate::utils::*;
+
+#[derive(Args)]
+pub struct InviteDevice {
+    /// Parsec config directory
+    #[arg(short, long)]
+    config_dir: Option<PathBuf>,
+    /// Device slughash
+    #[arg(short, long)]
+    device: Option<String>,
+}
+
+pub async fn invite_device(invite_device: InviteDevice) -> anyhow::Result<()> {
+    let config_dir = invite_device.config_dir.unwrap_or(get_default_config_dir());
+    let devices = list_available_devices(&config_dir).await;
+
+    if let Some(device_slughash) = invite_device.device {
+        let mut possible_devices = vec![];
+
+        for device in &devices {
+            if device.slughash().starts_with(&device_slughash) {
+                possible_devices.push(device);
+            }
+        }
+
+        match possible_devices.len() {
+            0 => {
+                println!("Device `{device_slughash}` not found, available devices:");
+                format_devices(&devices);
+            }
+            2 => {
```

Comment:
Shouldn't this match equal or greater than 2 devices?

Reply:
True, I was drunk this day :joy: